src/Microsoft.AspNet.OData.Shared/Formatter/Serialization/SelectExpandNode.cs

```diff
@@ -182,24 +206,115 @@ public SelectExpandNode(SelectExpandClause selectExpandClause, IEdmStructuredTyp
+            foreach (SelectItem selectItem in selectedItems)
             {
                 ExpandedNavigationSelectItem expandItem = selectItem as ExpandedNavigationSelectItem;
                 if (expandItem != null)
                 {
-                    ValidatePathIsSupported(expandItem.PathToNavigationProperty);
+                    ValidatePathIsSupportedForExpand(expandItem.PathToNavigationProperty);
                     NavigationPropertySegment navigationSegment = (NavigationPropertySegment)expandItem.PathToNavigationProperty.LastSegment;
                     IEdmNavigationProperty navigationProperty = navigationSegment.NavigationProperty;
                     if (allNavigationProperties.Contains(navigationProperty))
                     {
-                        ExpandedNavigationProperties.Add(navigationProperty, expandItem.SelectAndExpand);
+                        if (!ExpandedNavigationProperties.ContainsKey(navigationProperty))
+                        {
+                            ExpandedNavigationProperties.Add(navigationProperty, expandItem.SelectAndExpand);
+                        }
+                        else
+                        {
+                            ExpandedNavigationProperties[navigationProperty] = expandItem.SelectAndExpand;
+                        }
+                    }
+                    else
+                    {
+                        //This is the case where the navigation property is not on the current type. We need to propagate the expand item to deeper SelectExpandNode.
+                        IEdmProperty complexProperty = FindNextPropertySegment(expandItem.PathToNavigationProperty);
+
+                        if (complexProperty != null && ExpandedNavigationPropertiesOnComplexTypes.ContainsKey(complexProperty))
+                        {
+                            SelectExpandClause oldClause = ExpandedNavigationPropertiesOnComplexTypes[complexProperty];
+                            SelectExpandClause newClause = new SelectExpandClause(oldClause.SelectedItems.Concat(new List<SelectItem>() { expandItem }), false);
+                            ExpandedNavigationPropertiesOnComplexTypes.Add(complexProperty, newClause);
+                        }
+                        else if (complexProperty != null)
+                        {
+                            List<SelectItem> expandedItemsForClause = new List<SelectItem>();
+                            expandedItemsForClause.Add(expandItem);
+                            SelectExpandClause expandItems = new SelectExpandClause(expandedItemsForClause, false);
+                            ExpandedNavigationPropertiesOnComplexTypes.Add(complexProperty, expandItems);
+                        }
                     }
                 }
             }
         }
 
+        /// <summary>
+        /// Finds the appropriate property segment which should be responsible for propagating the expand item.
```

Comment:
>Finds the appropriate property segment which should be responsible for propagating the expand item. [](start = 12, length = 99)

can you add an example into the summary to illustrate the functionality of this method for better maintenance later.